src/rpc/rpctx.cpp

```diff
@@ -586,8 +586,11 @@ Value registercontracttx(const Array& params, bool fHelp)
 
     RPCTypeCheck(params, list_of(str_type)(str_type)(int_type)(int_type)(str_type));
 
-    string luaScriptFilePath = params[1].get_str();
-    if (luaScriptFilePath.find("/tmp/lua") == std::string::npos)
+    string luaScriptFilePath = GetAbsolutePath(params[1].get_str()).string();
+    if (luaScriptFilePath.empty())
+        throw JSONRPCError(RPC_SCRIPT_FILEPATH_NOT_EXIST, "Lua Script file not exist!");
+
+    if (luaScriptFilePath.find("/tmp/lua") != 0)
```

Comment:
should use the startswith("/tmp/lua") method
example:
    const std::string TEMP_LUA_DIR = "/tmp/lua";
    if (luaScriptFilePath.compare(0, TEMP_LUA_DIR.size(), TEMP_LUA_DIR) != 0) {